iroh-bytes/src/baomap.rs

```diff
@@ -181,19 +181,38 @@ pub trait Store: ReadableStore + PartialMap {
     ///
     /// Returns the hash of the imported file. The reason to have this method is that some database
     /// implementations might be able to import a file without copying it.
-    fn import(
+    fn import_file(
         &self,
         data: PathBuf,
         mode: ImportMode,
         format: BlobFormat,
         progress: impl ProgressSender<Msg = ImportProgress> + IdGenerator,
     ) -> BoxFuture<'_, io::Result<(TempTag, u64)>>;
 
-    /// This trait method imports data from memory.
+    /// Import data from memory.
     ///
     /// It is a special case of `import` that does not use the file system.
     fn import_bytes(&self, bytes: Bytes, format: BlobFormat) -> BoxFuture<'_, io::Result<TempTag>>;
 
+    /// Import data from a stream of bytes.
+    fn import_stream(
```

Comment:
would it make sense for the non file based imports to allow adding a name manually?

Reply:
This always adds raw blobs which don't have names. Adding tags and wrapping collections happens at the `node.rs` layer.

What we could do is add a`WrapOption` to the RPC message and client interface which would be 
```rust
enum WrapOption {
  NoWrap,
  Wrap { name: String }
```
which would allow to wrap the single blob in a collection with a single entry by passing the name here. Do we want/need this?

We might want to add a client & rpc API to create collections from a list of hashes instead maybe? Also not sure.

Reply:
I meant to be able to tag the raw blobs, is that not sth we want to do?

Reply:
That is possible already: The [client API](https://github.com/n0-computer/iroh/pull/1550/files#diff-481403521f6c5ced02ca5150086fc28be9ad2f8e0e518c3742222f3c8b504b84R276) has a `tag` option, and it is handled in the [rpc handler in `node.rs`](https://github.com/n0-computer/iroh/pull/1550/files#diff-4bd93a0bc8c0813c38e9d897deb531fef9ca4517415c3f3f873a9cb3dbefa595R1412)